nemo/collections/speechlm/modules/modality_adapter.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+from dataclasses import dataclass
+from typing import Optional
+
+from megatron.core.model_parallel_config import ModelParallelConfig
+from megatron.core.transformer.module import MegatronModule
+from megatron.core.transformer.transformer_config import TransformerConfig
+
+from nemo.collections.asr.modules import ConformerEncoder
+from nemo.collections.speechlm.utils import to_dict_config
+from nemo.core.classes.common import Serialization
+from nemo.core.classes.module import NeuralModule
+from nemo.lightning import io
+
+
+class MCoreModalityAdapterModule(MegatronModule):
```

Comment:
Can you add a docstring describing what this is?

Reply:
done

Reply:
just to follow nemo2 convention